llvm/lib/AsmParser/LLParser.cpp

```diff
@@ -3530,7 +3530,12 @@ bool LLParser::parseTargetExtType(Type *&Result) {
   if (parseToken(lltok::rparen, "expected ')' in target extension type"))
     return true;
 
-  Result = TargetExtType::get(Context, TypeName, TypeParams, IntParams);
+  auto TTy =
+      TargetExtType::getOrError(Context, TypeName, TypeParams, IntParams);
+  if (auto E = TTy.takeError())
+    return tokError(toString(std::move(E)));
```

Comment:
Would be good to do this in BitcodeReader as well.

Reply:
Done, but I don't know how to write a test case for error handling in BitcodeReader.

Reply:
That's fine, we don't usually test bitcode error handling.